src/ImageSharp.Web/Providers/PhysicalFileSystemProvider.cs

```diff
@@ -43,19 +42,10 @@ public class PhysicalFileSystemProvider : IImageProvider
 #endif
             FormatUtilities formatUtilities)
         {
+            Guard.NotNull(options, nameof(options));
             Guard.NotNull(environment, nameof(environment));
 
-            // ContentRootPath is never null.
-            // https://github.com/dotnet/aspnetcore/blob/b89eba6c3cda331ee98063e3c4a04267ec540315/src/Hosting/Hosting/src/WebHostBuilder.cs#L262
-            Guard.NotNullOrWhiteSpace(environment.WebRootPath, nameof(environment.WebRootPath));
-
-            // Allow configuration of the provider without having to register everything
-            PhysicalFileSystemProviderOptions providerOptions = options != null ? options.Value : new();
-            string providerRootPath = GetProviderRoot(providerOptions, environment.WebRootPath, environment.ContentRootPath);
-
-            // Ensure provider directory is created before initializing the file provider
-            Directory.CreateDirectory(providerRootPath);
-            this.fileProvider = new PhysicalFileProvider(providerRootPath);
+            this.providerRootPath = GetProviderRoot(options.Value, environment.WebRootPath, environment.ContentRootPath);
             this.formatUtilities = formatUtilities;
         }
```

Comment:
As mentioned in https://github.com/SixLabors/ImageSharp.Web/pull/225#discussion_r802467562, this would disable the provider if the provider root directory doesn't exist:

```suggestion

            // Disable provider if root path doesn't exist
            if (!Directory.Exists(this.providerRootPath))
            {
                this.Match = _ => false;
            }
        }
```

Reply:
No, that’s not wise as described above. If my provider isn’t matching with no explanation that’s an afternoon of debugging wasted

Reply:
OK, so it should be a little more nuanced: when a specific root provider path is configured that doesn't exist, it should try to create that directory (like [ASP.NET also does](https://github.com/dotnet/aspnetcore/blob/c85baf8db0c72ae8e68643029d514b2e737c9fae/src/Hosting/Hosting/src/Internal/HostingEnvironmentExtensions.cs#L51)), which would then throw an exception if it's invalid...

Having the default `IImageProvider` throwing exceptions when you've not done any customizations to ImageSharps configuration would be a bad experience in my opinion.

Reply:
You've missed my point I think. Hopefully my additional explanation above makes sense.

I do agree that the provider should try to create the directory if it is explicitly set and doesn't exist.